File: PO/Project/C++/src/hogweed.cpp
#include "hogweed.h"
#include <vector>

hogweed::hogweed(YX position, world* world_point)
: plant(10, position ,world_point)
{
    avatar = 'h';
}

int hogweed::Action()
{
    if (rand() % 50 == 0) {
        Reproduct();
        //return 0;
    }

    //kill neighbours
    std::vector<field*> near = world_point->GetFieldsNear(position);

    for (int i = 0; i < near.size(); i++) 
    {
        if (near[i]->member != nullptr) 
        {
            world_point->GetLogger()->Log("hogweed killed an organizm with the area attak");
            world_point->Destroy(near[i]->member);
        }
    }
    
    return 1;
}

bool hogweed::RejectAttak(organizm* attaker)
{
    attaker->IAmEddarOfHouseStartkLordOfWinterfellAndWardenOfTheNorthSentenceYouToDie();

    world_point->GetLogger()->Log("hogweed instantly killed an attaker");

    return true;
}

organizm* hogweed::MakeChild(YX new_position)
{
    return (new hogweed(new_position, world_point));
}

hogweed::~hogweed()
{
}

File: PO/Project/JavaProject/src/Plants/Hogweed.java
package Plants;

import Abstractions.*;
import java.awt.*;
import java.util.List;
import java.util.Random;

public class Hogweed extends Plant {

    public Hogweed(World world,Point position) {
        super(10, position, world);
        this.avatar = 'h';
        this.image_path = "Hogweed.png";
        loadImage();
    }

    @Override
    public int action() {
        if (new Random().nextInt(50) == 0) {
            reproduct();
            // return 0;
        }

        // Kill neighbors
        List<Field> near = world.getFieldsNear(position);

        for (Field field : near) {
            if (field.getMember() != null && !(field.getMember() instanceof Hogweed)) {
                world.getLogger().log("hogweed killed an organism with the area attack");
                world.destroy(field.getMember());
            }
        }

        return 1;
    }

    @Override
    public boolean rejectAttack(Organism attacker) {
        attacker.die();
        world.getLogger().log("hogweed instantly killed an attacker");
        return true;
    }

    @Override
    public Organism makeChild(Point newPosition) {
        return new Hogweed(world, newPosition);
    }
}

